OneSignalSDK/onesignal/src/main/java/com/onesignal/influence/OSInAppMessageTracker.java

```diff
@@ -95,6 +95,7 @@ void addSessionData(@NonNull JSONObject jsonObject, OSInfluence influence) {
 
     @Override
     public void cacheState() {
-        dataRepository.cacheIAMInfluenceType(influenceType == null ? OSInfluenceType.UNATTRIBUTED : influenceType);
+        OSInfluenceType influenceTypeToCache = influenceType == null ? OSInfluenceType.UNATTRIBUTED : influenceType;
+        dataRepository.cacheIAMInfluenceType(influenceTypeToCache == OSInfluenceType.DIRECT ? OSInfluenceType.INDIRECT : influenceTypeToCache);
     }
```

Comment:
A comment should be added to explain that within `cacheState` for IAM tracker we always downgrade `DIRECT` since leaving the app while showing an IAM would be a `DIRECT` state